### PR TITLE
Upd

### PR DESCRIPTION
**Pull request by dandimog**

### CODE REVIEW

**Review by JuliaNortman — APPROVED** [review]

**Review by JuliaNortman — CHANGES_REQUESTED** [review]
Cannot be merged automatically
You should resolve conflicts

File: marketplace-client/src/marketplace/src/app/account/forgot-password/forgot-password.component.ts [hunk 1]
```diff
@@ -49,8 +49,11 @@ export class ForgotPasswordComponent{
         },
         error: error => {
           console.log(error);
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/account/forgot-password/forgot-password.component.ts:51]: don't commit such sensitive logs

File: marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+</div> -->
+
+
+      <div class="">
+        <label>Status:</label>
+
+        <div *ngFor="let status of status_list" class="px-2">
+          <div class="custom-control custom-radio">
+            <input type="radio" class="custom-control-input" value={{status}} name="isActive" id={{status}}
+              [checked]="activeRadioButton == status" (change)="onChange($event)">
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.html]: `===` works faster
https://codeburst.io/javascript-double-equals-vs-triple-equals-61d4ce5a121a

File: marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  </div>
+</div> -->
+
+
+      <div class="">
+        <label>Status:</label>
+
+        <div *ngFor="let status of status_list" class="px-2">
+          <div class="custom-control custom-radio">
+            <input type="radio" class="custom-control-input" value={{status}} name="isActive" id={{status}}
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.html]: it's more readable to write 1 attr = 1 line

File: marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<!--<div>
+  <label for="isActive">Filter by:</label>
+
+  <div *ngFor="let status of status_list">
+    <input id={{status}} type="radio" value={{status}} name="isActive" [checked]="activeRadioButton == status"
+      (change)="onChange($event)">
+    <label for={{status}}>{{status}}</label>
+  </div>
+</div> -->
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.html]: don't commit commented out code (at least you can add TODO with a description if you plan to use it later)

File: marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.html [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+
+      <div class="">
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.html]: better add some class here to make CSS selectors more specific

File: marketplace-client/src/marketplace/src/app/system-accounts/manager-list/manager-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+import { StaffMember } from 'src/app/_models/staff-member';
+
+@Component({
+  selector: 'app-manager-list',
+  templateUrl: './manager-list.component.html',
+  styleUrls: ['./manager-list.component.css'],
+})
+export class ManagerListComponent {
+  @Input() managers: StaffMember[] = [];
+  readonly status_list: string[] = ['all', 'active', 'terminated'];
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/system-accounts/manager-list/manager-list.component.ts]: by convention, js var names are in camelCase

File: marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  templateUrl: './filter.component.html',
+  styleUrls: ['./filter.component.css'],
+})
+export class FilterComponent implements OnInit {
+  activeRadioButton = '';
+  users: StaffMember[] = [];
+  @Input() status_list: string[] = [];
+
+  //@Output() results: EventEmitter<User[]> = new EventEmitter<User[]>();
+  @Output() results: EventEmitter<any> = new EventEmitter<any>();
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.ts:24]: specify the type

File: marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+  ngOnInit(): void {
+    this.activeRadioButton = this.getFilter();
+    this.filterStatus(this.activeRadioButton, true);
+  }
+
+  filterStatus(status: string, init: boolean) {
+    this.service
+      .getFilteredUsers(status, init)
+      .subscribe((results: UserDto) => {
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.ts:40]: don't forget to unsubscribe

File: marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.ts [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        this.users = results.users;
+        this.results.emit();
+        this.service.pageNumSource.next(results.pageNum);
+        this.service.pageSource.next(results.currentPage);
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.ts]: we get some info from the service and then send it back. The service can do it itself. If we can delegate some data-processing work from a component to a service, then better do it. The component should be as small as possible and do only UI-rendering work

File: marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.ts [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  @Output() results: EventEmitter<any> = new EventEmitter<any>();
+
+  constructor(private service: SystemAccountService) {}
+
+  ngOnInit(): void {
+    this.activeRadioButton = this.getFilter();
+    this.filterStatus(this.activeRadioButton, true);
+  }
+
+  filterStatus(status: string, init: boolean) {
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/system-accounts/filter/filter.component.ts]: all your class members are public. But some of them are used only inside of the class and might be made private

File: marketplace-client/src/marketplace/src/app/system-accounts/search/search.component.ts [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+  users: StaffMember[] = [];
+  search: string = '';
+  init: boolean = true;
+
+  constructor(private service: SystemAccountService, private el: ElementRef) {}
+
+  ngOnInit(): void {
+    this.search = this.service.getSearch();
+
+    fromEvent(this.el.nativeElement, 'keyup')
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/system-accounts/search/search.component.ts]: it's safer and faster not to listen to `keyup` events, but to utilize angular's built-in event emitters. Any HTML input has a bunch of outputs from js-native events, like `(change)`, `(click)`, `(blur)`, etc. You can just use 1-way output binding to listen to them
(the current option is not an error, as well)

File: marketplace-client/src/marketplace/src/app/account/_components/password.form.component.ts [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+  ) {
+    this.form = this.formBuilder.group({
+      password: ['', [Validators.required, Validators.minLength(6)]],
+      confirmPassword: ['', Validators.required],
+    }, {
+      validator: [validateConfirmPassword, validatePassword]
+    });
+  }
+
+  get getForm(): { [p: string]: AbstractControl } { return this.form.controls; }
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/account/_components/password.form.component.ts:38]: `Record<string, AbstractControl>`

https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype

File: marketplace-client/src/marketplace/src/app/account/_components/password.form.component.ts [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+    }
+    this.loading = true;
+    this.form.disable();
+    this.route.queryParamMap.subscribe(params => {
+      const id = params.get('link');
+      console.log(id);
+      if (id) {
+        this.accountService.setNewPassword(id, this.form.get('password')?.value)
+          .pipe(first())
+          .subscribe({
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/account/_components/password.form.component.ts:53]: `subscribe` inside of the `subscribe` (moreover, without unsubscriptions) is a source of memory leaks and unexpected async errors. Leverage the RxJS flattening operators:

https://ramya-bala221190.medium.com/simple-example-for-rxjs-flattening-operators-2456628e4c07

http://alanpryorjr.com/2019-05-15-rxjs-flattening-operators/#:~:text=In%20reality%2C%20the%20flattening%20operators,observable%20that%20is%20emitting%20values.&text=Contrast%20this%20with%20map%20%2C%20which,value%20(not%20another%20observable).

File: marketplace-client/src/marketplace/src/app/system-accounts/register-stuff/register-stuff.component.ts [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+
+  courierRoleSelected(): boolean {
+    return this.form.value.role == Role.Courier;
+  }
+
+  pmRoleSelected(): boolean {
+    return this.form.value.role == Role.ProductManager;
+  }
+
+  private mapToStaffMember(o: any): StaffMember {
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/system-accounts/register-stuff/register-stuff.component.ts:67]: argument types

File: marketplace-client/src/marketplace/src/app/_services/system-account.service.ts [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+      url = url + '/manager';
+    } else {
+      url = url + '/courier';
+    }
+    return this.http.get<UserDto>(url, {
+      params: this.buildQueryParams(filter, search, page),
+    });
+  }
+
+  buildQueryParams(filter: string, search: string, page: number): HttpParams {
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/_services/system-account.service.ts]: don't forget about access modifiers

**Review by MaxxMorozov — COMMENTED** [review]
- what's the PR about? the name isn't informative, no description
- in general, the PR is too large. Looks like just "all my work from the last week". PR should be as small as possible - 1 atomic piece of work, 1 feature, 1 bugfix, 1 refactoring, etc.
- I've left 1 type of each comment just FYI. there's a bunch of issues of the same type throughout the PR

**Pull request closed without merging**